tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,838 @@
+        logger.info(f"Updating initialDelaySeconds on {mon.name} deployment")
+        insert_delay(mon_dep=mon.name)
+
+    logger.info("Sleeping for mons to get initialized")
+    time.sleep(60)
+    logger.info("Validating whether all mons reached running state")
+    mon_pods = get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+    for mon in mon_pods:
+        wait_for_resource_state(resource=mon, state=constants.STATUS_RUNNING)
+
+
+def corrupt_ceph_monitors():
+    """
+    Corrupts ceph monitors by deleting store.db file
+
+    """
+    mon_pods = get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+    for mon in mon_pods:
+        logger.info(f"Corrupting monitor: {mon.name}")
+        mon_id = mon.get().get("metadata").get("labels").get("ceph_daemon_id")
+        logger.info(
+            _exec_cmd_on_pod(
+                cmd=f"rm -rf /var/lib/ceph/mon/ceph-{mon_id}/store.db", pod_obj=mon
+            )
+        )
+        try:
+            wait_for_resource_state(resource=mon, state=constants.STATUS_CLBO)
+        except ResourceWrongStatusException:
+            if (
+                mon.ocp.get_resource(resource_name=mon.name, column="STATUS")
+                != constants.STATUS_CLBO
+            ):
+                logger.info(
+                    f"Re-spinning monitor: {mon.name} since it did not reach CLBO state"
+                )
+                mon.delete()
+    logging.info("Validating all the monitors are in CLBO state")
+    for mon in get_mon_pods(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE):
+        wait_for_resource_state(resource=mon, state=constants.STATUS_CLBO)
+
+
+def archive_and_mute_ceph_warn():
+    """
+    Archives all ceph crashes and mute warnings
+
+    """
+    toolbox = pod.get_ceph_tools_pod()
+    toolbox.exec_ceph_cmd(ceph_cmd="ceph crash archive-all")
+    toolbox.exec_ceph_cmd(
+        ceph_cmd="ceph config set mon auth_allow_insecure_global_id_reclaim false"
+    )
+
+
+def ceph_fs_recovery():
+    """
+    Resets the CephFS
+
+    """
+    toolbox = pod.get_ceph_tools_pod()
+    try:
+        toolbox.exec_cmd_on_pod(
+            f"ceph fs reset {defaults.CEPHFILESYSTEM_NAME} --yes-i-really-mean-it"
+        )
+    except CommandFailed:
+        toolbox.exec_cmd_on_pod(
+            f"ceph fs new {defaults.CEPHFILESYSTEM_NAME} ocs-storagecluster-cephfilesystem-metadata "
+            f"ocs-storagecluster-cephfilesystem-data0 --force"
+        )
+        toolbox.exec_cmd_on_pod(
+            f"ceph fs reset {defaults.CEPHFILESYSTEM_NAME}  --yes-i-really-mean-it"
+        )
+
+
+def get_ceph_caps(secret_resource):
+    """
+    Gets ocs secrets and decodes it to get ceph keyring
+
+    Args:
+        secret_resource (any): secret resource name
+
+    """
+    keyring = ""
+
+    fs_node_caps = """
+        caps mds = "allow rw"
+        caps mgr = "allow rw"
+        caps mon = "allow r"
+        caps osd = "allow rw tag cephfs *=*"
+"""
```

Comment:
indent issue

Reply:
This indentation is based on trail and error. When i tab the last line of the string it messes up the ceph auth when I try to import the locally written file with all the ceph caps to ceph.